common/src/Polyhedron_Matcher.h

```diff
@@ -254,23 +248,26 @@ class PolyhedronMatcher {
      * @param delta the move delta
      * @return the vertex relation
      */
-    static VertexRelation buildVertexRelation(const P& left, const P& right, const typename V::Set& vertices, const V& delta) {
-        typename V::Map vertexMap;
-        
+    static VertexRelation buildVertexRelation(const P& left, const P& right, typename V::List vertices, const V& delta) {
+        VMap vertexMap;
+
+        VectorUtils::setCreate(vertices);
+
         auto* firstVertex = left.vertices().front();
         auto* currentVertex = firstVertex;
         do {
             const auto& position = currentVertex->position();
-            if (vertices.count(position) == 0) {
-                if (right.hasVertex(position))
+            if (VectorUtils::setContains(vertices, position)) {
```

Comment:
To emphasize that this is an exact search, I might add to the method comment:

`vertices` are expected to be exact positions of vertices in `left`, whereas the vertex positions searched for in `right` allow an epsilon of `Math::Constants<T>::almostZero()`

Reply:
done